packages/material-ui/src/styles/experimentalStyled.test.js

```diff
@@ -49,44 +49,60 @@ describe('experimentalStyled', () => {
   });
 
   describe('muiOptions', () => {
-    const theme = createMuiTheme({
-      components: {
-        MuiTest: {
-          variants: [
-            {
-              props: { variant: 'rect', size: 'large' },
-              style: {
-                width: '400px',
-                height: '400px',
+    /**
+     * @type {ReturnType<typeof createMuiTheme>}
+     */
+    let theme;
+    /**
+     * @type {ReturnType<typeof styled>}
+     */
+    let Test;
+
+    before(() => {
+      theme = createMuiTheme({
+        components: {
+          MuiTest: {
+            variants: [
+              {
+                props: { variant: 'rect', size: 'large' },
+                style: {
+                  width: '400px',
+                  height: '400px',
+                },
+              },
+            ],
+            styleOverrides: {
+              root: {
+                width: '250px',
+              },
+              rect: {
+                height: '250px',
               },
-            },
-          ],
-          styleOverrides: {
-            root: {
-              width: '250px',
-            },
-            rect: {
-              height: '250px',
             },
           },
         },
-      },
-    });
-
-    const testOverridesResolver = (props, styles) => ({
-      ...styles.root,
-      ...(props.variant && styles[props.variant]),
+      });
+
+      const testOverridesResolver = (props, styles) => ({
+        ...styles.root,
+        ...(props.variant && styles[props.variant]),
+      });
+
+      expect(() => {
+        Test = styled(
+          'div',
+          { shouldForwardProp: (prop) => prop !== 'variant' && prop !== 'size' },
+          { muiName: 'MuiTest', overridesResolver: testOverridesResolver },
+        )`
+          width: 200px;
+          height: 300px;
+        `;
+      }).toErrorDev([
+        'You have illegal escape sequence in your template literal',
+        'You have illegal escape sequence in your template literal',
+      ]);
```

Comment:
@mnajdova Any idea why this is happening?

Reply:
Nice! I was wondering where this warning was coming from!

Reply:
I am not sure why it is happening. Let's replace the string template with object:
```
({
  width: '200px',
  height: '300px',
})  
```

and it will go away. I will try to repro it on a simple example and report it.

Reply:
Update, even just adding round bracket around it solves the issue 🤷  :
```
(`
  width: 200px;
  height: 300px;
`)
```

Reply:
https://codesandbox.io/s/objective-vaughan-6r3u0?file=/src/App.js

Found the issue, seems like the destructuring on line 31 causes the warning:
```
return defaultStyledResolver(...stylesWithDefaultTheme);
```
If I remove it the warning is gone.
```
return defaultStyledResolver(stylesWithDefaultTheme);
```
 However, as far as I remember `styled-components` was not working without the destructuring. Will check this again to confirm

Reply:
Sounds like this is what the adapter packages were made for.

Reply:
True, I will try to update them today and test again

Reply:
Marked as FIXME. Needs to be fixed before release. For now we keep the expectation so that we don't add noise to the console output.